nilearn/plotting/__init__.py

```diff
@@ -3,26 +3,31 @@
 """
 # Authors: Chris Filo Gorgolewski, Gael Varoquaux
 
+from ..version import (_import_module_with_version_check,
+                       OPTIONAL_MATPLOTLIB_MIN_VERSION)
+
 ###############################################################################
 # Make sure that we don't get DISPLAY problems when running without X on
 # unices
 
-def _set_mpl_backend():
-    # We are doing local imports here to avoid poluting our namespace
-    import os
-    if not os.name == 'posix':
-        # We have the problem only on posix systems
-        return
-    if 'DISPLAY' in os.environ:
-        return
 
+def _set_mpl_backend():
     try:
+        # We are doing local imports here to avoid poluting our namespace
         import matplotlib
-        # Agg is a backend that will do PNGs and PDFs
-        matplotlib.use('Agg')
+        import os
+        # We have the problem only on posix systems
+        if os.name == 'posix' and 'DISPLAY' not in os.environ:
+            # Agg is a backend that will do PNGs and PDFs
+            matplotlib.use('Agg')
     except ImportError:
-        pass
         # No need to fail here, eg during tests
+        pass
+    else:
+        # When matplotlib was successfully imported we need to check
+        # that the version is greater that the minimum required one
+        _import_module_with_version_check('matplotlib',
+                                          OPTIONAL_MATPLOTLIB_MIN_VERSION)
```

Comment:
Yeah LooseVersion doesn't seem to be very knowledgeable about rc, alpha, et al. versions ...

```
In [1]: from distutils.version import LooseVersion

In [2]: LooseVersion('1.1.1rc') > LooseVersion('1.1.1')
Out[2]: True

In [3]: LooseVersion('1.1.1.a1') > LooseVersion('1.1.1')
Out[3]: True
```

Also we decided to drop support for bog-standard Ubuntu 12.04 versions long ago in #397 by requiring sklearn >= 0.12.1.